src/reporters/reporter-util.ts

```diff
@@ -0,0 +1,209 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { fileURLToPath } from 'url';
+import { resolve, sep } from 'path';
+import { readFileSync } from 'fs';
+import { inspect } from 'util';
+import { gray } from 'kleur/colors';
+import { codeFrameColumns } from '@babel/code-frame';
+import StackUtils from 'stack-utils';
+import { Location, StackFrame, TestError } from '../common_types';
+
+const stackUtils = new StackUtils({ internals: StackUtils.nodeInternals() });
+
+const SYNTHETICS_PATH = resolve(__dirname, '..', '..');
+const PW_CORE_PATH = require.resolve('playwright-core');
+
+function prepareStackFrame(line: string): StackFrame {
+  const frame = stackUtils.parseLine(line);
+  if (!frame) {
+    return;
+  }
+  // ignore node internals
+  if (frame.file?.startsWith('internal') || frame.file?.startsWith('node:')) {
+    return;
+  }
+
+  // handle relative URLs
+  let file = frame.file?.startsWith('file://')
+    ? fileURLToPath(frame.file)
+    : resolve(process.cwd(), frame.file);
+
+  // ignore node_modules
+  if (file.includes(`${sep}node_modules${sep}`)) {
+    return;
+  }
+
+  // filter library and PW files
+  if (!filterLibInternals(file)) {
+    return;
+  }
+
+  // When we bundle the journeys, we store the absolute path of the journey
+  // files and write the sourcemap relative to these files. When we try to
+  // extract the sourcemap file location, the stacktrace will be relatively
+  // resolved to these files which would be under `journeys` folder. So we strip
+  // these extra `journeys` from the path to get the correct file location.
+  if (frame.file?.includes('journeys/journeys')) {
+    file = file.replace('journeys/journeys', 'journeys');
+  }
+
+  return {
+    file,
+    line: frame.line || 0,
+    column: frame.column || 0,
+    function: frame.function,
+  };
+}
+
+function filterLibInternals(file: string) {
+  // To signore filtering the stack trace on tests
+  if (process.env.TEST_OVERRIDE) {
+    return true;
+  }
+  if (file.startsWith(SYNTHETICS_PATH)) return false;
```

Comment:
nit: I would use the same convention for all the `if` blocks with one statement (like we do in line 84):

```
if (file.startsWith(SYNTHETICS_PATH)) { 
  return false;
}

 // should have been filtered by node_modules, but just in case
if (file.startsWith(PW_CORE_PATH)) {
  return false;
}
```